From: Yifeng <wuyifeng@nd.com.cn>
Date: Mon, 18 Oct 2021 07:01:00 +0000
Subject: Set default page url

The requirement is that it should load to the default url when the following cases:
* when Chromium opens.
* when the button of home is clicked.
* when a new tab is opened.
AP9-2162 Change the default url to http://myPromethean.com.
AP9-5158 Change the default url to https://www.prometheanworld.com/.
AP9-10977 AP9-10970 Fix the issue that the home page settings do not work.

---
 chrome/android/java/src/org/chromium/chrome/browser/homepage/HomepageManager.java | 20 ++--
 chrome/android/java/src/org/chromium/chrome/browser/tabmodel/ChromeTabCreator.java | 2 +-
 2 files changed, 11 insertions(+), 11 deletions(-)

diff --git a/chrome/android/java/src/org/chromium/chrome/browser/homepage/HomepageManager.java b/chrome/android/java/src/org/chromium/chrome/browser/homepage/HomepageManager.java
--- a/chrome/android/java/src/org/chromium/chrome/browser/homepage/HomepageManager.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/homepage/HomepageManager.java
@@ -148,16 +148,16 @@ public class HomepageManager implements HomepagePolicyManager.HomepagePolicyStat
      */
     public static String getDefaultHomepageUri() {
         String homepagePartnerDefaultUri;
-        String homepagePartnerDefaultGurlSerialized =
-                SharedPreferencesManager.getInstance().readString(
-                        ChromePreferenceKeys.HOMEPAGE_PARTNER_CUSTOMIZED_DEFAULT_GURL, "");
-        if (!homepagePartnerDefaultGurlSerialized.equals("")) {
-            homepagePartnerDefaultUri =
-                    GURL.deserialize(homepagePartnerDefaultGurlSerialized).getSpec();
-        } else {
+        // String homepagePartnerDefaultGurlSerialized =
+        //         SharedPreferencesManager.getInstance().readString(
+        //                 ChromePreferenceKeys.HOMEPAGE_PARTNER_CUSTOMIZED_DEFAULT_GURL, "");
+        // if (!homepagePartnerDefaultGurlSerialized.equals("")) {
+        //     homepagePartnerDefaultUri =
+        //             GURL.deserialize(homepagePartnerDefaultGurlSerialized).getSpec();
+        // } else {
             homepagePartnerDefaultUri = SharedPreferencesManager.getInstance().readString(
-                    ChromePreferenceKeys.HOMEPAGE_PARTNER_CUSTOMIZED_DEFAULT_URI, "");
-        }
+                    ChromePreferenceKeys.HOMEPAGE_PARTNER_CUSTOMIZED_DEFAULT_URI, "https://www.prometheanworld.com/");
+        // }
         if (!homepagePartnerDefaultUri.equals("")) return homepagePartnerDefaultUri;

         return UrlConstants.NTP_NON_NATIVE_URL;
@@ -228,7 +228,7 @@ public class HomepageManager implements HomepagePolicyManager.HomepagePolicyStat
      *
      */
     public boolean getPrefNTPIsHomepageEnabled() {
-        return mSharedPreferencesManager.readBoolean(ChromePreferenceKeys.HOMEPAGE_NTP_IS_HOMEPAGE, false);
+        return mSharedPreferencesManager.readBoolean(ChromePreferenceKeys.HOMEPAGE_NTP_IS_HOMEPAGE, true);
     }

     /**
diff --git a/chrome/android/java/src/org/chromium/chrome/browser/tabmodel/ChromeTabCreator.java b/chrome/android/java/src/org/chromium/chrome/browser/tabmodel/ChromeTabCreator.java
--- a/chrome/android/java/src/org/chromium/chrome/browser/tabmodel/ChromeTabCreator.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/tabmodel/ChromeTabCreator.java
@@ -298,7 +298,7 @@ public class ChromeTabCreator extends TabCreator {
      * @return the created tab.
      */
     public Tab launchUrl(String url, @TabLaunchType int type, Intent intent, long intentTimestamp) {
-        if (!mIncognito && url.equals(UrlConstants.NTP_URL)) {
+        if (url.equals(UrlConstants.NTP_URL)) {
             if (HomepageManager.getInstance().getPrefNTPIsHomepageEnabled()) {
                 url = HomepageManager.getInstance().getHomepageUri();
             }
--
2.17.1